File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/observer/lab/Developer.java
// Developer.java

public class Developer
{
	public Developer()
	{
		myMeetings = new DpVector<Meeting>();
	}

	public void add(Meeting aMeeting)
	{
		if (aMeeting == null)
		{
			throw new DpException("void Developer.add(Meeting aMeeting) -- Invalid aMeeting.");
		}

		System.out.println(
			  "Developer.add("
			+ aMeeting.toString()
			+ ")"
			);
		myMeetings.add(aMeeting);
	}

	public void remove(Meeting aMeeting)
	{
		if (aMeeting == null)
		{
			throw new DpException("void Developer.remove(Meeting aMeeting) -- Invalid aMeeting.");
		}

		System.out.println(
			  "Developer.remove("
			+ aMeeting.toString()
			+ ")"
			);
		// I was too lazy to implement this. :-(
	}

	public void update(Meeting aMeeting)
	{
		if (aMeeting == null)
		{
			throw new DpException("void Developer.update(Meeting aMeeting) -- Invalid aMeeting.");
		}

		System.out.println(
			  "Developer.update("
			+ aMeeting.toString()
			+ ")"
			);
	}

	private DpVector<Meeting> myMeetings;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/observer/lab/developer.cpp
// developer.cpp

#include <iostream>
#include "dp_exception.h"
#include "dp_vector.h"
#include "developer.h"
#include "meeting.h"

void developer::add(meeting * a_meeting)
{
	if (a_meeting == nullptr)
	{
		throw dp_exception("void developer::add(meeting * a_meeting) -- Invalid a_meeting.");
	}

	std::cout
		<< "developer.add("
		<< a_meeting->to_string()
		<< ")"
		<< std::endl;
	my_meetings.add(a_meeting);
}

void developer::remove(meeting * a_meeting)
{
	if (a_meeting == nullptr)
	{
		throw dp_exception("void developer::remove(meeting * a_meeting) -- Invalid a_meeting.");
	}

	std::cout
		<< "developer.remove("
		<< a_meeting->to_string()
		<< ")"
		<< std::endl;
	// I was too lazy to implement this. :-(
}

void developer::update(meeting * a_meeting)
{
	if (a_meeting == nullptr)
	{
		throw dp_exception("void developer::update(meeting * a_meeting) -- Invalid a_meeting.");
	}

	std::cout
		<< "developer.update("
		<< a_meeting->to_string()
		<< ")"
		<< std::endl;
}
